pkg/extension/extensiontests/spec.go

```diff
@@ -0,0 +1,165 @@
+package extensiontests
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+)
+
+func (specs ExtensionTestSpecs) Walk(walkFn func(*ExtensionTestSpec)) ExtensionTestSpecs {
+	for i := range specs {
+		walkFn(specs[i])
+	}
+
+	return specs
+}
+
+func (specs ExtensionTestSpecs) Run(w *ResultWriter) error {
+	var results ExtensionTestResults
+
+	specs.Walk(func(spec *ExtensionTestSpec) {
```

Comment:
I was expecting that run-tests always tries to run things in parallel. If the tests can't run in parallel, they should share a mutex.

Reply:
I haven't quite got there yet.

It appears one of the reasons openshift-tests executes tests the way it does, is you can't run them in parallel with the weird way we invoke ginkgo tests.  I get a lot of strange behavior, and haven't figured out all the places I need mutexes.

Reply:
In-process parallelization of ginkgo specs is impossible afaik. I was biased toward ensuring other frameworks like gotest can work efficiently here. 
That said, I didn't think a default parallelism in run-test would be incongruous with ginkgo - I would think a mutex in BuildExtensionTestSpecsFromOpenShiftGinkgoSuite's definition of `Run: func() *ext.ExtensionTestResult{ }` provides for serialization.

Reply:
Oh, well that's easy enough do.

https://github.com/openshift-eng/openshift-tests-extension/pull/2/commits/324101173c14677cf2915a3611433fc9fa288cbb#diff-c5823b26ff6d2a83ced7fc032357f03398d38edb69adaa8488bab83d4b543f4eR52-R88